azext_iot/digitaltwins/providers/resource.py

```diff
@@ -97,6 +119,7 @@ def rbac_handler(lro):
                             )
 
             create_or_update.add_done_callback(rbac_handler)
+            create_or_update.add_done_callback(check_state)
```

Comment:
Do the callbacks run serially (assuming so)? We would want to run check_state prior to rbac_handler since rbac_handler assumes the resource has been fully provisioned.

Reply:
![image](https://user-images.githubusercontent.com/73560279/133318936-c0f9a4b3-f67c-44ff-9152-b372c6c18035.png)
will switch